spec/ic.did

```diff
@@ -43,6 +62,15 @@ service ic : {
   delete_canister : (record {canister_id : canister_id}) -> ();
   deposit_cycles : (record {canister_id : canister_id}) -> ();
   raw_rand : () -> (blob);
+  http_request : (record {
+    url : text;
+    method : variant { get };
+    headers: vec http_header;
+    body : opt blob;
+    transform : opt variant {
+      function: func (http_response) -> (http_response) query
```

Comment:
I am not sure how exactly that is implemented. Is this safe for canister upgrades?

Reply:
@Dfinity-Bjoern What do you think would be unsafe here in case of an upgrade?

To my knowledge a `func` in candid is represented by a principal + a method name, and only the canister's own principal would be accepted here. If a canister is upgraded and the method being referenced here is, for example, removed, then the call to the transform will fail.